### PR TITLE
Implement dialog element for dialogs with accessibility

### PR DESCRIPTION
**Pull request by t03857785**
## References
#6579 Modal dialogs are not accessible

## Code changes

Change the dialog to use the dialog html element rather than the div element
Add aria title to the dialog
Add aria modal to the dialog to control focus for accessibility tools using aria
Add aria title to buttons and implement where required (this may be expanded upon in the future)

## User-facing changes

Users using accessibility tools will now have dialog labels announced and more meaningful titles for some dialog buttons 
Other users will not see any changes

## Backwards-incompatible changes

Dialog buttons now have an optional ariaLabel that can be used as well as label if the label is not suitable for users with accessibility needs

### CODE REVIEW

File: packages/apputils/src/sessioncontext.tsx [hunk 1]
```diff
@@ -1410,14 +1416,20 @@ export class SessionContextDialogs implements ISessionContext.IDialogs {
       throw new Error('No kernel to restart');
     }
 
-    const restartBtn = Dialog.warnButton({ label: 'Restart' });
+    const restartBtn = Dialog.warnButton({
+      label: 'Restart',
```
Comment (krassowski) [inline packages/apputils/src/sessioncontext.tsx:1420]: It seems it was missing previously, let's add it in:

```suggestion
      label: trans.__('Restart'),
```

**Comment by t03857785** [comment]
This was tested on windows using the built in Narrator

**Comment by krassowski** [comment]
The visual test failures are relevant - tests need to be updated by adding [`exact: true`](https://playwright.dev/docs/api/class-locator#locator-get-by-role-option-exact) to the options of affected `getByRole` calls.

<details>

Example issue:

```
Error: locator.click: Error: strict mode violation: locator('.jp-Dialog').getByRole('button', { name: 'Select' }) resolved to 2 elements:
    1) <button class="jp-Dialog-button jp-mod-reject jp-mod-sty…>…</button> aka getByRole('button', { name: 'Cancel Kernel Select' })
    2) <button class="jp-Dialog-button jp-mod-accept jp-mod-sty…>…</button> aka getByRole('button', { name: 'Select Kernel' })
```

</details>

**Comment by t03857785** [comment]
@krassowski I think all of the required/requested changes have been made I have also on reflection reverted the No Kernel Aria Label to fix tests as it may not be accurate in all situations

**Review by krassowski — APPROVED** [review]
Looks good to me, thank you @t03857785.

**Pull request merged**